core/src/main/java/nl/nn/adapterframework/util/DB2DocumentWriter.java

```diff
@@ -0,0 +1,298 @@
+/*
+   Copyright 2022 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package nl.nn.adapterframework.util;
+
+import java.sql.JDBCType;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+import org.apache.logging.log4j.Logger;
+import org.xml.sax.SAXException;
+
+import nl.nn.adapterframework.core.SenderException;
+import nl.nn.adapterframework.jdbc.dbms.IDbmsSupport;
+import nl.nn.adapterframework.stream.MessageOutputStream;
+import nl.nn.adapterframework.stream.StreamingException;
+import nl.nn.adapterframework.stream.document.ArrayBuilder;
+import nl.nn.adapterframework.stream.document.DocumentBuilderFactory;
+import nl.nn.adapterframework.stream.document.DocumentFormat;
+import nl.nn.adapterframework.stream.document.INodeBuilder;
+import nl.nn.adapterframework.stream.document.ObjectBuilder;
+
+public class DB2DocumentWriter {
+	protected static Logger log = LogUtil.getLogger(DB2DocumentWriter.class);
+
+	private String docname = "result";
+	private String recordname = "rowset";
+	private String nullValue = "";
+	private boolean trimSpaces=true;
+	private boolean decompressBlobs=false;
+	private boolean getBlobSmart=false;
+	private String blobCharset = Misc.DEFAULT_INPUT_STREAM_ENCODING;
+	private static boolean convertFieldnamesToUppercase = AppConstants.getInstance().getBoolean("jdbc.convertFieldnamesToUppercase", false);
+
+	public static String getFieldType (int type) {
```

Comment:
moet dit niet private en getFieldName zijn?

Reply:
nee, dit is niet de naam van het veld, maar de naam van datatype